mypy/plugins/enums.py

```diff
@@ -78,6 +78,22 @@ class SomeEnum:
     """
     enum_field_name = _extract_underlying_field_name(ctx.type)
     if enum_field_name is None:
+        # We do not know the enum field name (perhaps it was passed to a function and we only
+        # know that it _is_ a member).  All is not lost however, if we can prove that the all
+        # of the enum members have the same value-type, then it doesn't matter which member
+        # was passed in.  The value-type is still known.
+        if isinstance(ctx.type, Instance):
+            info = ctx.type.type
+            stnodes = (info.get(name) for name in info.names)
+            first_node = next(stnodes, None)
+            if first_node is None:
+                return ctx.default_attr_type
+            first_node_type = first_node.type
+            if all(node is not None and node.type == first_node_type for node in stnodes):
+                underlying_type = get_proper_type(first_node_type)
+                if underlying_type is not None:
+                    return underlying_type
```

Comment:
If I read the docstring of `get_proper_type()` correctly, it suggest that here you should actually be returning `first_node_type`. And in fact, perhaps you should probably adjust the `all()` check to go through `get_proper_type()` instead?

Reply:
I took a stab at this, but I'm not completely sure what you're saying here :).  It's unclear to me whether the result of `enum_value_callback` should be a `ProperType` or not.  The existing code seems to try to return a `ProperType` so I just followed suit there.  I did change the "all-equal" to compare the entities after figuring out their `ProperType` though.

Reply:
Hm, I think I misunderstood this. I thought that using `get_proper_type()` would distinguish `A` from `int` if we have
```
A = int
def f(a: A): ...
```
but it doesn't seem to work that way. So you're forgiven for not understanding me. :-)

I think what you have now is fine.